From 15a24eca9c605948fe3e5fc9061c6b5962ff3d46 Mon Sep 17 00:00:00 2001
From: Erik Stromdahl <erik.stromdahl@gmail.com>
Date: Thu, 17 Aug 2017 18:40:15 +0200
Subject: [PATCH] fix issue: invalid SSL_version specified at ...

The SSL version string passed to IO::Socket::SSL->start_SSL
does not work with more recent versions of the io-socket-ssl
perl module.
Changing it to just TLSv1_2 solves the issue...

Signed-off-by: Erik Stromdahl <erik.stromdahl@gmail.com>
---
 sendEmail | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sendEmail b/sendEmail
index 9f9392e..b7146fa 100755
--- a/sendEmail
+++ b/sendEmail
@@ -1903,7 +1903,7 @@ else {
     if ($conf{'tls_server'} == 1 and $conf{'tls_client'} == 1 and $opt{'tls'} =~ /^(yes|auto)$/) {
         printmsg("DEBUG => Starting TLS", 2);
         if (SMTPchat('STARTTLS')) { quit($conf{'error'}, 1); }
-        if (! IO::Socket::SSL->start_SSL($SERVER, SSL_version => 'SSLv3 TLSv1')) {
+        if (! IO::Socket::SSL->start_SSL($SERVER, SSL_version => 'TLSv1_2')) {
             quit("ERROR => TLS setup failed: " . IO::Socket::SSL::errstr(), 1);
         }
         printmsg("DEBUG => TLS: Using cipher: ". $SERVER->get_cipher(), 3);
-- 
2.13.3

